lib/HLSL/HLOperationLowerExtension.cpp

```diff
@@ -999,6 +991,14 @@ class CustomLowering
                         }
                     }
                 }
+                else
+                {
+                    // This also disables 'exploding' of vectors
```

Comment:
I think the comment here should be something like

Do this after checking for the vector index to allow the vector to be manually exploded.

Reply:
Yep, saw that on a re-review just after I pushed.